storage/innobase/trx/trx0trx.cc

```diff
@@ -1534,6 +1536,41 @@ trx_commit_in_memory(
 	}
 }
 
+/*************************************************************//**
+Update the persistent counts of all tables modified by a transaction. */
+void trx_update_persistent_counts(
+/*============*/
+    trx_t* trx)		/*!< in: transaction */
+{
+    dict_table_t* ib_table;
+    TABLE* table;
+    bool update_persistent_count;
+    for (trx_mod_tables_t::const_iterator t = trx->mod_tables.begin();
+         t != trx->mod_tables.end(); t++) {
+        ib_table = t->first;
+
+        table = thd_get_open_tables(current_thd);
```

Comment:
Why do we need this call?

`current_thd` should be avoided whenever possible. Here, `trx->mysql_thd` should be set. It is `NULL` for system transactions that update the InnoDB data dictionary tables, which are not visible to the SQL layer anyway.

Reply:
> Why do we need this call?
I'm presuming you're referring to why there's a separate `thd_get_open_tables(thd)` call instead of just `thd->open_tables`. `trx0trx.cc` includes `trx0trx.h` and eventually forward declares `THD`, but as a result, even if I include `sql_class.h` (where `THD` is defined), the forward declaration is found first, so none of `THD`'s members or functions can be accessed.